src/tests/roc_core/test_mov_histogram.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2023 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <CppUTest/TestHarness.h>
+
+#include "roc_core/heap_arena.h"
+#include "roc_core/mov_histogram.h"
+
+namespace roc {
+namespace core {
+
+namespace {
+
+enum { NumObjects = 10, EmbeddedCap = 5 };
+
+struct Object {
+    static long n_objects;
+
+    size_t value;
+
+    Object(size_t v = 0)
+        : value(v) {
+        n_objects++;
+    }
+
+    Object(const Object& other)
+        : value(other.value) {
+        n_objects++;
+    }
+
+    ~Object() {
+        n_objects--;
+    }
+};
+
+long Object::n_objects = 0;
+
+} // namespace
+
+TEST_GROUP(movhistogram) {
+    HeapArena arena;
+};
+
+TEST(movhistogram, single_pass) {
+    const size_t value_range_min = 0;
+    const size_t value_range_max = 100;
+    const size_t num_bins = 10;
+    const size_t win_length = 10;
+
+    MovHistogram<size_t> hist(arena, value_range_min, value_range_max, num_bins,
+                              win_length);
+    CHECK(hist.is_valid());
+
+    size_t values[win_length];
```

Comment:
What is the purpose of this value? I think, you can eliminate it here and in the second test.

Reply:
The `values` array used to store the added values that we add to the Histogram, but I think it's better to remove it.